include/term-slug.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * @since 3.7
+ */
+class PLL_Term_Slug {
+
+	/**
+	 * @var PLL_Translated_Term
+	 */
+	private $translated_term;
+
+	/**
+	 * @var PLL_Language
+	 */
+	private $lang;
+
+	/**
+	 * @var int
+	 */
+	private $parent;
+
+	/**
+	 * @var string
+	 */
+	private $taxonomy;
+
+	/**
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * @var string
+	 */
+	private $name;
+
+	/**
+	 * Constructor
+	 *
+	 * @since 3.7
+	 *
+	 * @param PLL_Translated_Term $translated_term Term translation object.
+	 * @param string              $slug            The term slug.
+	 * @param string              $taxonomy        The term taxonomy.
+	 * @param string              $name            The term name.
+	 */
+	public function __construct( PLL_Translated_Term $translated_term, string $slug, string $taxonomy, string $name ) {
+		$this->translated_term = $translated_term;
```

Comment:
Yep, same thing, but we missed the use of this class for SQL queries.

However, is there any point in keeping this code in `PLL_Model`?

Reply:
We discussed it earlier and agree on using `PLL_Translated_Term` for the following reasons:
- It goes in the right direction regarding single responsibility of `PLL_Model`, especially if we want to merge `PLL_Amin_Model` and `PLL_Model` and reduce its size.
- Type is stronger using `PLL_Translated_Term` instead of `PLL_Model`.

There is another solution but it'd imply to create static methods in `PLL_Translated_Object` (which seems feasible because a lot of properties are hardcoded by child classes). This way we'd have only `_construct( string $slug, string $name, string $taxonomy )`.